astromodels/functions/template_model.py

```diff
@@ -834,3 +952,106 @@ def to_table_model(self, file_name, model_name, overwrite=False):
             tmf.add_interpolation_data(self._spectrum[i, :], **input_dict)
 
         tmf.save_data(overwrite=overwrite)
+
+
+
+def convert_old_table_model(model_name: str):
+    from pandas import HDFStore
+
+    # Get the data directory
+
+    data_dir_path: Path = get_user_data_path()
+
+    # Sanitize the data file
+    
+    filename_sanitized = data_dir_path / f"{model_name}.h5"
+    
+    
+    if not filename_sanitized.exists():
+
+        log.error("The data file %s does not exists. Did you use the "
+             "TemplateFactory?" % (filename_sanitized))
+         
+        raise MissingDataFile( )
+    
+    with HDFStore(filename_sanitized) as store:
+    
+        data_frame = store["data_frame"]
+
+        parameters_grids = collections.OrderedDict()
+
+        processed_parameters = 0
+
+        for key in list(store.keys()):
+
+            match = re.search("p_([0-9]+)_(.+)", key)
+            
+            if match is None:
+                
+                continue
+            
+            else:
+                
+                tokens = match.groups()
+                
+                this_parameter_number = int(tokens[0])
+                this_parameter_name = str(tokens[1])
+                
+                assert (
```

Comment:
maybe change to log.error and AssertionError?

Reply:
done!